DiscordChatExporter.Core/Exporting/ChannelExporter.cs

```diff
@@ -39,6 +39,10 @@ public async ValueTask ExportChannelAsync(ExportRequest request, IProgress<doubl
             var encounteredUsers = new HashSet<User>(IdBasedEqualityComparer.Instance);
             await foreach (var message in _discord.GetMessagesAsync(request.Channel.Id, request.After, request.Before, progress))
             {
+                // Skips any messages that fail to pass the supplied filter
+                if (!request.MessageFilter.Filter(message))
+                    continue;
+
```

Comment:
I chose to put the logic here instead of in the writer with the partitioning logic to avoid instantiating the writer/file if no messages are being exported (and the "exportedAnything" flag is here).

Reply:
Makes sense :+1: